File: read_docx.js
const mammoth = require("mammoth");
const fs = require("fs");

const filePath = process.argv[2];

if (!filePath) {
    console.error("Please provide a file path");
    process.exit(1);
}

mammoth.extractRawText({ path: filePath })
    .then(function (result) {
        const text = result.value;
        fs.writeFileSync("docx-content.txt", text);
        console.log("Extracted text to docx-content.txt");
    })
    .catch(function (error) {
        console.error(error);
    });

File: read_docx.py
from docx import Document
import sys

def extract_text_from_docx(file_path):
    try:
        doc = Document(file_path)
        full_text = []
        for index, para in enumerate(doc.paragraphs):
            full_text.append(para.text)
        print('\n'.join(full_text))
    except Exception as e:
        print(f"Error: {e}")

if __name__ == '__main__':
    if len(sys.argv) > 1:
        extract_text_from_docx(sys.argv[1])
    else:
        print("Please provide a file path")
